### PR TITLE
Better docs -- module reference

### PR DESCRIPTION
**Pull request by hanke**
This is quite basic, except for the handling of datalad.api. Don't think sphinx-autosummary has been used for something like that.

Issue 1) we cannot easily get a call signature for those class instances -- because they are class instances. It may be possible to make classes from datalad.interfaces look like functions in datalad.api, but this need some vodka and a whole night.

Issue 2) Currently we practice numpydoc conventions. If we do not want to give that up (don't think we should) we need python-numpydoc in order to build the docs (hence needs to go into requirements.txt)

### CODE REVIEW

**Comment by yarikoptic** [comment]
is it possible to specify an alternative requirements.txt?  due to its inflexibility I already for a while thought to create requirements-dev.txt which would have absorbed all those needed for testing and now docs.

**Comment by yarikoptic** [comment]
and since it is WiP I guess it wasn't yet for the merge, so I guess I will not merge it atm, but then not sure how/what to review beyond spell checking and nodding  ;)

**Comment by hanke** [comment]
Numpydoc issue is resolved. Sphinx now comes with a 'napoleon' extension that supports it natively. Will push later.

**Comment by hanke** [comment]
No longer WiP

File: datalad/customremotes/archive.py [hunk 1]
```diff
@@ -256,7 +255,6 @@ def req_CHECKPRESENT(self, key):
         TODO: just proxy the call to annex for underlying tarball
 
         Replies
-        -------
```
Comment (yarikoptic) [inline datalad/customremotes/archive.py]: Does offer confuse the beast? Could we use some other level of describing here

File: datalad/support/configparserinc.py [hunk 2]
```diff
@@ -17,36 +17,36 @@
 
 if sys.version_info >= (3,2): # pragma: no cover
 
-	# SafeConfigParser deprecated from Python 3.2 (renamed to ConfigParser)
```
Comment (yarikoptic) [inline datalad/support/configparserinc.py]: Now will become hard to pick up fixes from fail2ban's copy :-(
Was it screaming out loud about tabs? But OK let's do it for uniformity

**Comment by yarikoptic** [comment]
Can't merge from the phone because of bloody failing windows... Go ahead

File: datalad/customremotes/archive.py (hunk 1 above)
Comment (hanke) [inline datalad/customremotes/archive.py]: Here is the list of supported headings:

http://sphinx-doc.org/latest/ext/napoleon.html#docstring-sections

I wasn't sure what "repies" refers to.

File: datalad/support/configparserinc.py (hunk 2 above)
Comment (hanke) [inline datalad/support/configparserinc.py]: Yeah, felt sad, but also looked ugly ;-)

**Pull request merged**